src/main/java/work/multithreaded/Bank.java

```diff
@@ -1,58 +1,65 @@
 package work.multithreaded;
 
 import java.math.BigInteger;
-import java.util.Random;
+import java.util.Objects;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ThreadLocalRandom;
 
 public class Bank {
     private final int numberOfAccounts;
     private final long minBalance;
     private final long maxBalance;
-    private final Random random = new Random();
+
     private final ConcurrentHashMap<Integer, Long> accountsBalances = new ConcurrentHashMap<>();
 
     public Bank(int numberOfAccounts, long minBalance, long maxBalance) {
+        if (numberOfAccounts <= 0) {
+            throw new IllegalArgumentException("numberOfAccounts must be > 0");
+        }
+        if (minBalance > maxBalance) {
+            throw new IllegalArgumentException("minBalance must be <= maxBalance");
+        }
         this.numberOfAccounts = numberOfAccounts;
         this.minBalance = minBalance;
         this.maxBalance = maxBalance;
 
         long range = maxBalance - minBalance + 1;
         for (int i = 0; i < numberOfAccounts; i++) {
-            long balance = minBalance + (Math.abs(random.nextLong()) % range);
+            long delta = ThreadLocalRandom.current().nextLong(range);
+            long balance = minBalance + delta;
             accountsBalances.put(i, balance);
         }
     }
 
     public int pickRandomAccountId() {
-        return random.nextInt(2000);
+        return ThreadLocalRandom.current().nextInt(numberOfAccounts);
```

Comment:
🙂 
How ThreadLocal works in virtual threads? What are alternatives? What can go wrong?